vignettes/generic.Rmd

```diff
@@ -18,363 +18,490 @@ library(admiraldev)
-are set to `NA`. Observations in the additional dataframe which have no match in
-the original dataframe are ignored.
-
-## Examples
-
-A simple call to derive `ADSL` randomization date (`ADSL.RANDDT`) from an
-intermediate additional dataframe would use `derive_vars_merged()` as follows.
-
-```{r eval=TRUE}
-# Use DM domain as basis to build ADSL
-adsl_01 <- dm %>%
-  select(-DOMAIN)
-
-# Convert disposition character date to numeric date and
-# join as randomization date to ADSL
-adsl_02 <- adsl_01 %>%
-  derive_vars_merged(
-    dataset_add = ds,
-    filter_add = DSDECOD == "RANDOMIZED",
-    by_vars = exprs(STUDYID, USUBJID),
-    new_vars = exprs(RANDDT = convert_dtc_to_dt(DSSTDTC))
-  )
-```
 
-This call would return the input dataframe with the variable `RANDDT` added.
-
-Now, an example to add first treatment datetime (`ADSL.TRTSDT`), where we
-need to make use of `derive_vars_merged()` with the `mode` and `order` arguments
-to select the required observations from the additional dataframe. Note: the
-`filter_add` argument here shows a possible method for checking only for valid doses.
-
-```{r eval=TRUE}
-# Convert exposure start date to numeric date without imputation,
-# determine first exposure datetime and add to ADSL
-adsl_03 <- adsl_02 %>%
-  derive_vars_merged(
-    dataset_add = ex,
-    filter_add = (EXDOSE > 0 | (EXDOSE == 0 & str_detect(EXTRT, "PLACEBO"))) &
-      !is.na(TRTSDT),
-    new_vars = exprs(TRTSDT = convert_dtc_to_dt(EXSTDTC)),
-    order = exprs(TRTSDT, EXSEQ),
-    mode = "first",
-    by_vars = exprs(STUDYID, USUBJID)
-  )
-```
+# Characterization of Derivations
+
+The generic functions can be characterized by the following three properties:
+
+- What should be added? There are two options:
+    - _Variables_
+    - _Records_
+- What is the source data? There are two options:
+    - A _single_ source dataset
+    - _Multiple_ source datasets
+- Which method should be used? There are three options:
+    - _Selection_: The new values are derived by selecting records, e.g., the
+    baseline records, the last exposure record, ...
+    - _Summary_: The new values are derived by summarizing values, e.g., sum,
+    average, concatenation, ...
+    - _Computation_: The new values are derived by a computation with more than
+    one value as input, e.g., deriving BMI or BSA from height and weight 
+
+## Overview of Derivations
+
+Using the three properties makes it easy to find the appropriate function for a
+particular derivation. The following interactive table lists all generic
+functions and their properties.
+
+```{r, echo=FALSE}
+library(reactable)
+
+generic_derivations <- tibble::tribble(
+  ~Derivation,                      ~Method,       ~What,       ~Source,
+  "derive_var_extreme_flag()",      "selection",   "variables", "single",
+  "derive_var_joined_exist_flag()", "selection",   "variables", "single",
+  "derive_var_merged_exist_flag()", "selection",   "variables", "single",
```

Comment:
add the function from https://github.com/pharmaverse/admiral/pull/2232?

Reply:
Yes, I have to update it once the PR (and some other PRs) are merged.

Reply:
The order of the bullet points doesn't line up with the order of the columns.